xerparser/validation.py

```
@@ -0,0 +1,713 @@
+    }
+
+    # Valid duration types
+    VALID_DURATION_TYPES: Set[str] = {
+        'DT_FixedDrtn', 'DT_FixedUnits', 'DT_FixedDUR2', 'DT_FixedRate'
+    }
+
+    # Valid constraint types
+    VALID_CONSTRAINT_TYPES: Set[str] = {
+        'CS_MEO', 'CS_MEOA', 'CS_MSO', 'CS_MSOA', 'CS_MSOB',
+        'CS_ALAP', 'CS_MANDSTART', 'CS_MANDFIN', None, ''
+    }
```

Ruff (0.14.13)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)